### PR TITLE
Update tests, implement refresh for other sdks

### PR DESCRIPTION
**Pull request by msub2**
This PR makes a long needed update to the SDK tests, removing tests for old styles and adding new ones for Prebid mechanisms, as well as adding tests for the react-three-fiber SDK. It also adds support for ad refreshing to the SDKs that were missing it (sans Web, which doesn't have any meaningful usage currently). Lastly, it updates the workflows for GH Actions to build the r3f and WLE test projects in CI, fixing #206.

### CODE REVIEW

File: tests/aframe.test.mjs
Comment (limbofeather) [inline tests/aframe.test.mjs:91]: Would these tests continue to be relevant if we're not using iframes?

File: tests/web.test.mjs
Comment (limbofeather) [inline tests/web.test.mjs:58]: Again - I think these wouldn't be necessary once we subset support for 2D.

File: tests/aframe.test.mjs
Comment (limbofeather) [inline tests/aframe.test.mjs:103]: Do you think if we're repeating the url/image pairing again and again, they could be a shared constant across the test files?

File: threejs/src/index.js
Comment (limbofeather) [inline threejs/src/index.js:74]: Is this a very different `checkVisibility` compared to what we have for aframe?

File: r3f/src/ZestyBanner.js
Comment (limbofeather) [inline r3f/src/ZestyBanner.js:59]: Hmm any way we could stop repeating this function across multiple files? Would be great if we could consolidate.

File: tests/babylonjs.test.mjs
Comment (limbofeather) [inline tests/babylonjs.test.mjs:79]: Why did we halve the timeout?

File: tests/web.test.mjs
Comment (msub2) [reply tests/web.test.mjs:12]: Yeah, I think in the grand scheme of things there's not much benefit to the web SDK over regular web ads.

File: tests/aframe.test.mjs
Comment (msub2) [reply tests/aframe.test.mjs:91]: Yes, I'm injecting an iframe here to replicate getting an actual ad, since the prebid wrapper can't fetch real ones for testing.

File: tests/aframe.test.mjs
Comment (msub2) [reply tests/aframe.test.mjs:103]: Good point yeah, I'll move them to some constants.

File: tests/babylonjs.test.mjs
Comment (msub2) [reply tests/babylonjs.test.mjs:79]: Initially I set it to 15 to match the refresh interval, but since the first simulated ad actually gets picked up in the first few seconds I realized I could lower it to 8 seconds, inject a new ad, and then check again after another 8 seconds, which would be one second after a refresh occurs.

File: tests/aframe.test.mjs
Comment (limbofeather) [reply tests/aframe.test.mjs:103]: Thanks 👍🏼

**Comment by msub2** [comment]
I think I'll actually try and make the consolidated visibility check function use the A-Frame visibility check also. Initially I used the dot product method from Wonderland since it seemed more consistent to implement across frameworks (and it didn't involve reimplementing the three.js frustum classes and bounding box intersection calculations), but if it's localized to one function then I don't need to worry about split implementations

**Comment by limbofeather** [comment]
> I think I'll actually try and make the consolidated visibility check function use the A-Frame visibility check also. Initially I used the dot product method from Wonderland since it seemed more consistent to implement across frameworks (and it didn't involve reimplementing the three.js frustum classes and bounding box intersection calculations), but if it's localized to one function then I don't need to worry about split implementations

Yeah this was my main thought as well - if it's in three or four different places it might be difficult to keep track of which implementation we're using if logically they're the same.

File: utils/helpers.js
Comment (limbofeather) [inline utils/helpers.js:180]: Do we want to keep this log?

File: tests/test-constants.mjs
Comment (limbofeather) [inline tests/test-constants.mjs:1]: Thank you 🙏🏼

**Review by limbofeather — APPROVED** [review]
One small comment but otherwise LGTM.

File: utils/helpers.js
Comment (msub2) [reply utils/helpers.js:180]: Oh whoops, missed that one, lemme get rid of that.

File: tests/babylonjs.test.mjs
Comment (limbofeather) [inline tests/babylonjs.test.mjs:70]: Might be useful to evaluate if the ad is linking out to the proper url.

File: tests/test-constants.mjs
Comment (limbofeather) [inline tests/test-constants.mjs:2]: Can't do `export const`?

File: threejs/src/index.js
Comment (limbofeather) [inline threejs/src/index.js:50]: Feel this entire section could be broken out to a fn, something like `checkIfBeingViewedForRefresh` so it's clear to understand what's happening here:

```javascript
setInterval(() => {
  checkIfBeingViewedForRefresh(this);
}, AD_REFRESH_INTERVAL);
```

WDYT?

File: threejs/src/index.js
Comment (limbofeather) [inline threejs/src/index.js:65]: Do we still need this `console.log`?

**Review by limbofeather — APPROVED** [review]
A few comments but we should be good to merge right after 🚀

File: threejs/src/index.js
Comment (msub2) [reply threejs/src/index.js:65]: Another one that slipped through, removed

File: tests/test-constants.mjs
Comment (msub2) [reply tests/test-constants.mjs:2]: Done

File: threejs/src/index.js
Comment (msub2) [reply threejs/src/index.js:50]: Yeah that sounds good, I'll split it out

**Pull request merged**
